SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp

```diff
@@ -0,0 +1,412 @@
+
+    Possible improvements to make:
+    money tracking? notifs tend to backlog.
+    find prize sprites - some code is there, just disabled
+    find translations for tera shards
+    other languages: make sure "bottle cap" isn't misread as "bottle of PP Up"
+    */
+
+    for(uint32_t c = 0; c < NUM_ROUNDS; c++) {
+        //Initiate dialog then mash until first battle starts
+        AdvanceDialogWatcher advance_detector(COLOR_YELLOW);
+        pbf_press_button(context, BUTTON_A, 10, 50);
+        int ret = wait_until(env.console, context, Milliseconds(7000), { advance_detector });
+        if (ret < 0) {
+            env.log("Dialog detected.");
+        }
+        pbf_mash_button(context, BUTTON_A, 300);
+        context.wait_for_all_requests();
+
+        NormalBattleMenuWatcher battle_menu(COLOR_YELLOW);
+        int ret_battle = run_until(
+            env.console, context,
+            [](BotBaseContext& context) {
+                pbf_mash_button(context, BUTTON_B, 10000); //it takes a while to load and start
+            },
+            { battle_menu }
+            );
+        if (ret_battle != 0) {
+            env.console.log("Failed to detect battle start!", COLOR_RED);
+        }
+        context.wait_for_all_requests();
+
+        bool battle_lost = false;
+        for (uint16_t battles = 0; battles < 4; battles++) {
+            NormalBattleMenuWatcher battle_menu2(COLOR_YELLOW); //Next battle started
+            OverworldWatcher overworld(COLOR_CYAN); //Previous battle was lost
+            
+            int ret_battle2 = run_until(
+                env.console, context,
+                [](BotBaseContext& context) {
+                    pbf_mash_button(context, BUTTON_B, 4000);
+                },
+                { battle_menu2, overworld }
+                );
+            context.wait_for_all_requests();
+
+            switch (ret_battle2) {
+            case 0:
+                env.log("Detected battle menu.");
+                run_battle(env, context);
+                break;
+            case 1:
+                env.log("Detected overworld.");
+                battle_lost = true;
+                stats.losses++;
+                env.update_stats();
+                send_program_status_notification(env, NOTIFICATION_STATUS_UPDATE);
+                break;
+            default:
+                env.log("Failed to detect battle menu or dialog prompt!");
+                stats.errors++;
+                env.update_stats();
+                send_program_status_notification(env, NOTIFICATION_STATUS_UPDATE);
+                break;
+            }
+
+            if (battle_lost) {
+                return_to_academy_after_loss(env, context);
+                break;
+            }
+        }
+
+        //Tournament won
+        if (!battle_lost) {
+            handle_end_of_tournament(env, context);
+        }
+
+        //Save the game if option is set
+        if (SAVE_NUM_ROUNDS != 0 && ((c + 1) % SAVE_NUM_ROUNDS) == 0) {
```

Comment:
`SAVE_NUM_ROUNDS` is a config option that's unlocked and can change at any time. So it's possible for it to be non-zero first, thus passing `SAVE_NUM_ROUNDS != 0`. Then it changes to zero before it does the `% SAVE_NUM_ROUNDS` which will lead to a crash.

Read it once and store to a variable before running this if-statement. That way both parts of the if-statement will be guaranteed to check the same value.